pkg/ccl/backupccl/restore_planning.go

```diff
@@ -797,15 +797,25 @@ func maybeUpgradeDescriptors(descs []catalog.Descriptor, skipFKsWithNoMatchingTa
 // "other" table is missing from the set provided are omitted during the
 // upgrade, instead of causing an error to be returned.
 func maybeUpgradeDescriptorsInBackupManifests(
-	backupManifests []backuppb.BackupManifest, skipFKsWithNoMatchingTable bool,
+	ctx context.Context,
+	backupManifests []backuppb.BackupManifest,
+	layerToIterFactory backupinfo.LayerToBackupManifestFileIterFactory,
+	skipFKsWithNoMatchingTable bool,
 ) error {
 	if len(backupManifests) == 0 {
 		return nil
 	}
 
+	// We do not need to upgrade descriptors that exist in the external SST
```

Comment:
I like this, and I'll probably delete this soon once we restrict restores to our new policy. In the meantime though can we check `backupManifests[0].ClusterVersion` instead?

Reply:
done, used cluster version.